purchase_work_acceptance/models/stock_picking.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    wa_id = fields.Many2one(
+        comodel_name='work.acceptance',
+        string='WA Reference',
+        domain=[('state', '=', 'accept')],
+    )
+
+    @api.multi
+    def button_validate(self):
+        self.ensure_one()
+        if self.wa_id:
+            wa_lines = self.env['work.acceptance.line']
+            for move in self.move_ids_without_package:
+                id = move.purchase_line_id.id
+                wa_line = wa_lines.search([('purchase_line_id', '=', id)])
+                if move.quantity_done > wa_line.product_qty:
+                    raise ValidationError(_('You cannot validate a transfer if'
+                                          ' Done more than accepted quantity'))
+        return super(Picking, self).button_validate()
+
+    @api.onchange('wa_id')
+    def wa_id_change(self):
+        if self.wa_id:
+            po_lines = self.env['purchase.order.line']
+            for move in self.move_ids_without_package:
+                id = move.purchase_line_id.id
+                po_line = po_lines.search([('id', '=', id)])
+                qty_limit = po_line.qty_accepted - po_line.qty_received
+                move.quantity_done = qty_limit
```

Comment:
NOTE: base on above comment about no link between po line and stock move, please rewrite this onchange.